### PR TITLE
jv_homework_StreamAPI

### PR DESCRIPTION
**Pull request by etezaadi**
Implemented all methods and class CandidateValidator.

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,7 +20,14 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        OptionalInt minIntValue = numbers.stream()
+                .map(i -> i.split(","))
+                .flatMap(Arrays::stream)
```
Comment (MaliukDaria) [inline src/main/java/practice/StreamPractice.java:25]: can we combine map and flatmap here into one flatmap?

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -23,7 +36,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 != 0)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```
Comment (MaliukDaria) [inline src/main/java/practice/StreamPractice.java:43]:
```suggestion
                .orElseThrow();
```

What exception orElseThrow() throws by default?

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -50,7 +70,12 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> ((p.getAge() >= fromAge && p.getAge() <= maleToAge)
+                        && Person.Sex.MAN == p.getSex())
+                        || (p.getAge() >= fromAge && p.getAge() <= femaleToAge)
+                        && Person.Sex.WOMAN == p.getSex())
```
Comment (MaliukDaria) [inline src/main/java/practice/StreamPractice.java:77]: can we use p.getAge() >= fromAge check only ones?

**Comment by etezaadi** [comment]
thanks for the comments :)

**Review by MaliukDaria — APPROVED** [review]

**Pull request closed without merging**